File: Python3/Medium/WrongAnswer/find-original-array-from-doubled-array_3.py
class Solution:
    def findOriginalArray(self, changed: List[int]) -> List[int]:
        
        if len(changed) % 2 == 1:
            return []
        changed.sort()
        original = []
        while changed:
            if changed[0] == 0:
                if len(changed) % 2 == 1:
                    return []
                else:
                    return original
            if changed[0] * 2 in changed:
                original.append(changed[0])
                changed.remove(changed[0])
                changed.remove(changed[0] * 2)
            else:
                return []
        return original

File: C++/Medium/WrongAnswer/find-original-array-from-doubled-array_3.cpp
class Solution {
public:
    vector<int> findOriginalArray(vector<int>& changed) {
        
        vector<int> res;
        sort(changed.begin(), changed.end());
        map<int, int> m;
        for (auto i: changed) {
            m[i]++;
        }
        for (auto i: changed) {
            if (m[i] == 0) continue;
            if (m.count(i*2) == 0 || m[i*2] == 0) return {};
            res.push_back(i);
            m[i]--;
            m[i*2]--;
        }
        return res;
    }
};